python/ray/_private/resource_spec.py

```diff
@@ -303,3 +324,20 @@ def _pretty_gpu_name(name):
         return None
     match = GPU_NAME_PATTERN.match(name)
     return match.group(1) if match else None
+
+
+def _get_mac_object_store_best_performance_capacity() -> int:
+    """Get the max object store capacity that doesn't suffer from performance
+    degration.
+
+    Mac with 16GB memory has degraded performance when object store size
+    is greater than 2GB.
+    (see https://github.com/ray-project/ray/issues/20388 for details)
+    The workaround here is to limit capacity to 2GB for 16GB Mac by default,
+    and raising warning if the capacity is overwritten by user.
+    """
+    system_memory = ray._private.utils.get_system_memory()
```

Comment:
Shall we just do it for all macs? For 8GB, the detected memory will be about 2GB anyways, so I think this would simplify the logic and prevent edge caess.